pkg/drainer/drainer.go

```diff
@@ -28,50 +28,59 @@ import (
 
 var drainHelper *drain.Helper
 var node *corev1.Node
+var nthConfig config.Config
 
-//InitDrainer will ensure the drainer has all the resources necessary to complete a drain
-func InitDrainer(nthConfig config.Config) {
+// InitDrainer will ensure the drainer has all the resources necessary to complete a drain.
+// This function must be called before any function in the drainer package.
+func InitDrainer(inputNthConfig config.Config) {
+	nthConfig = inputNthConfig
 	if drainHelper == nil {
-		drainHelper = getDrainHelper(nthConfig)
+		drainHelper = getDrainHelper()
 	}
-	node = &corev1.Node{}
 	var err error
-	if !nthConfig.DryRun {
-		node, err = drainHelper.Client.CoreV1().Nodes().Get(nthConfig.NodeName, metav1.GetOptions{})
-		if err != nil {
-			log.Fatalf("Couldn't get node %q: %s\n", nthConfig.NodeName, err.Error())
-		}
-		log.Printf("Successufully retrieved node: %s", node.Name)
+	node, err = FetchNode()
+	if err != nil {
+		log.Fatalf("Couldn't get node %q: %s\n", nthConfig.NodeName, err.Error())
+	}
+	log.Printf("Successufully retrieved node: %s", node.Name)
+}
+
+// FetchNode will send an http request to the k8s api server and return the corev1 model node
+func FetchNode() (*corev1.Node, error) {
+	node := &corev1.Node{}
+	if nthConfig.DryRun {
+		return node, nil
 	}
+	return drainHelper.Client.CoreV1().Nodes().Get(nthConfig.NodeName, metav1.GetOptions{})
 }
 
 //Drain will cordon the node and evict pods based on the config
-func Drain(nthConfig config.Config) {
+func Drain() {
 	if drainHelper == nil || node == nil {
 		InitDrainer(nthConfig)
 	}
 	var err error
 	if nthConfig.DryRun {
-		log.Printf("Node %s would have been cordoned, but dry-run flag was set", nthConfig.NodeName)
-	} else {
-		err = drain.RunCordonOrUncordon(drainHelper, node, true)
-		if err != nil {
-			log.Fatalf("Couldn't cordon node %q: %s\n", node, err.Error())
-		}
+		log.Printf("Node %s would have been cordoned and drained, but dry-run flag was set\n", nthConfig.NodeName)
+		return
 	}
+	cordonNode()
+	// Delete all pods on the node
+	err = drain.RunNodeDrain(drainHelper, nthConfig.NodeName)
+	if err != nil {
+		log.Fatalln(err.Error())
```

Comment:
Just something to do in a followup PR, not in this one...

Generally, it's best practice to never call `panic()` or the `FatalX` counterparts of the log package outside of the `main()` function. Instead, you want to return the error as-is (or wrapped using the pkg/errors `Wrap()` function) to the top-level entrypoint and have that entrypoint decide whether to segfault (panic) or print out the error to a log file or something and continue running.

See this excellent article from Dave Cheney about this topic: https://dave.cheney.net/tag/error-handling

Reply:
I went ahead and removed all panics from drainer in this PR since I was refactoring drainer anyways.  I think there are some more places this needs to be taken care of, but again, I'll include those changes in a cleanup PR after the sequence of scheduled maintenance events PRs get merged.